CONTRIBUTING.rst

```diff
@@ -3,45 +3,98 @@
+effort, and everyone is welcome to contribute. We value very much your feedback
+and opinion on features that should be improved or added. All discussions
+are public and hold on relevant issues or pull requests. To discuss your matter,
+please answer a relevant `issue <https://github.com/nilearn/nilearn/issues>`_
+or open a new one.
 
 The best way to contribute and to help the project is to start working on known
-issues.
-See `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ to get
-started.
+issues such as `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ ,
+`known bugs <https://github.com/nilearn/nilearn/labels/Bug>`_ or
+`proposed enhancements <https://github.com/nilearn/nilearn/labels/Enhancement>`_.
 
 If an issue does not already exist for a potential contribution, we ask that
 you first open an `issue <https://github.com/nilearn/nilearn/issues>`_ before
-sending a :ref:`pull request`.
+sending a :ref:`pull request` to discuss in advance scope and potential design
+choices.
+
+
+Contribution Guidelines
+---------------------------
+
+We ask every contributor to respect common guidelines. Those are inspired from
+`scikit-learn
+<https://scikit-learn.org/stable/developers/contributing.html#contributing-code>`_
+and ensure Nilearn remains simple to understand, efficient and maintanable.
+For example tests need to run quickly in order not to burden the development process.
+To keep continuous integration efficient with our limited infrastructure, running
+all the examples must lead to downloading a limited amount of data (gigabytes)
+and execute in a reasonable amount of time (a few hours).
+
+Those guidelines will hence be checked during reviewing process.
+
+
++--------------+-------------+---------------------------------------+
+|              | Which PR ?  |        Guidelines                     |
++==============+=============+=======================================+
+|              |             | - Clearly showcase benefits           |
+|  Examples    | New features| - Run in a few minutes                |
+|              |             | - Use light data from Nilearn         |
+|              |             | - Renders well after build            |
++--------------+-------------+---------------------------------------+
+|              |             | - Test type is adapted to behavior    |
+|              |             | - Tests pass continuous integration   |
+|              |  Bugfixes   | - Doesn't decrease coverage           |
+|    Tests     | New features| - Fast, using small mocked data       |
+|              |             | - Atomic (one per function) and seeded|
+|              |             | - For Bugfixes: non-regression test   |
++--------------+-------------+---------------------------------------+
+|              |             | - Variables, functions, arguments     |
+|              |             | have clear and consistent names       |
```

Comment:
This list renders oddly:

![image](https://user-images.githubusercontent.com/8228902/114210154-6dd22a00-992d-11eb-93fa-c6829c9c391d.png)

Reply:
Thank you very much for your quick feedback. Also this table is really not so beautiful with our current template. Not sure it's what we want to use. Any opinion ?

Reply:
A paragraph format might be easier to build, although it won't be as concise as a table... I don't have a strong opinion either way, unless it's not possible to make the table look good.

Reply:
do you think using something else than rst would help get the table right?

Reply:
I think it's mostly a style issue..